### PR TITLE
Create all pages and templates for header and footer.

### PR DESCRIPTION
**Pull request by andres-holguin**
This PR sets up all the planned pages for Andres's portfolio. Most of the content is placeholder text. No styling has been applied yet.

The header and footer are templates loaded dynamically to the DOM with fetch requests in the javascript file. The header has links to the other pages on the site. The footer has links to social media platforms.

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -26,3 +32,21 @@ function addRandomGreeting() {
   const greetingContainer = document.getElementById('greeting-container');
   greetingContainer.innerText = greeting;
 }
+
+// Fetch header for each page.
+fetch("templates/header.html")
+  .then(response => response.text())
```
Comment (nickKelman) [inline portfolio/src/main/webapp/script.js:38]: You can combine these then statements into one then statement.

Applies to below fetch statement as well.

Comment (andres-holguin) [reply portfolio/src/main/webapp/script.js:38]: Thanks for the tip, Nick! Here's what I was able to come up with.
```
  fetch("templates/header.html")
  .then( async response => {
    let header = document.createElement("header");
    header.innerHTML = await response.text();
    document.querySelector("body").before(header);
  });
```
Is this close to what you had in mind? Without waiting for the promise from response.text(), my header was only showing up as [HeaderObject] on the page.

Also, quick tip: When clicking the + sign to add a comment, you can also drag it to specifically comment on multiple lines of code.

Comment (nickKelman) [reply portfolio/src/main/webapp/script.js:38]: Thanks for the tip andres!

It seemed like in your first .then statement, you were calling 'response.text()' function and then passing that to a subsequent then block.

You should be able to do something like the following:

 .then(response => {
     const text = response.text();
     let footer = document.createElement("footer");
    footer.innerHTML = text;
    document.querySelector("body").after(footer);
 });

Comment (nickKelman) [reply portfolio/src/main/webapp/script.js:38]: If the return type of response.text is another promise, then the original way you coded it up is probably better than mixing promise notation with awaits.

**Pull request merged**